### PR TITLE
Doesn't exactly solve, but improves.

### PR DESCRIPTION
**Pull request by colinta**
This darn layout issue is tricky.

[Finishes #143618187](https://www.pivotaltracker.com/story/show/143618187)

### CODE REVIEW

File: Sources/Controllers/Stream/StreamViewController.swift [hunk 1]
```diff
@@ -546,9 +546,11 @@ final class StreamViewController: BaseElloViewController {
             guard let `self` = self else { return }
             self.imageCellHeightUpdated(streamImageCell)
         }
-        updateCellHeightNotification = NotificationObserver(notification: StreamNotification.UpdateCellHeightNotification) { [weak self] streamTextCell in
+        updateCellHeightNotification = NotificationObserver(notification: StreamNotification.UpdateCellHeightNotification) { [weak self] cell in
             guard let `self` = self else { return }
-            self.collectionView.collectionViewLayout.invalidateLayout()
+            nextTick {
```
Comment (colinta) [inline Sources/Controllers/Stream/StreamViewController.swift]: Helps, but didn't solve.

File: Sources/Controllers/Stream/Calculators/StreamImageCellSizeCalculator.swift [hunk 2]
```diff
@@ -92,7 +92,7 @@ class StreamImageCellSizeCalculator {
         if let assetWidth = imageBlock.asset?.gridLayoutAttachment?.width {
             imageWidth = min(imageWidth, CGFloat(assetWidth))
         }
-        return ceil((imageWidth / StreamImageCellSizeCalculator.aspectRatioForImageRegion(imageBlock)))
+        return ceil(imageWidth / StreamImageCellSizeCalculator.aspectRatioForImageRegion(imageBlock))
```
Comment (steam) [inline Sources/Controllers/Stream/Calculators/StreamImageCellSizeCalculator.swift]: Needs more parentheses. :trollface:

File: Sources/Controllers/Stream/Cells/StreamImageCell.swift [hunk 3]
```diff
@@ -183,10 +183,10 @@ class StreamImageCell: StreamRegionableCell {
     override func layoutSubviews() {
         super.layoutSubviews()
         if let aspectRatio = aspectRatio, let imageSize = imageSize {
-            let width = min(imageSize.width, self.frame.width - margin)
+            let width = min(imageSize.width, frame.width - margin)
             let actualHeight: CGFloat = ceil(width / aspectRatio) + Size.bottomMargin
-            if ceil(actualHeight) != ceil(frame.height) {
-                self.onHeightMismatch?(actualHeight)
+            if actualHeight != frame.height {
```
Comment (steam) [inline Sources/Controllers/Stream/Cells/StreamImageCell.swift]: Your change does _look_ correct but I swear this was triggering `onHeightMismatch` sometimes.

Comment (colinta) [inline Sources/Controllers/Stream/Cells/StreamImageCell.swift]: Oh yeah it definitely did at one point - we prob weren't using `ceil()` in the calculator.

**Pull request merged**